src/pytopmod/core/dcel/operators.py

```diff
@@ -0,0 +1,277 @@
+        for edge_key in list(face_trace(mesh, face_1_key, start_edge_key=edge_2_2_key)):
+            edge_node = mesh.edge_nodes[edge_key]
+            if edge_node.face_1_key == face_1_key:
+                edge_node.face_1_key = new_face_2_key
+            else:
+                edge_node.face_2_key = new_face_2_key
+
+        # Delete face_1 == face_2.
+        mesh.delete_face(face_1_key)
+
+
+def delete_edge(mesh: dcel_mesh.Mesh, old_edge: EdgeKey):
+    old_edge_node = mesh.edge_nodes[old_edge]
+
+    # 1.1 - Find the edges before and after the edge in the rotation of vertex_1.
+    vertex_1_rotation = list(vertex_trace(mesh, old_edge_node.vertex_1_key))
+    vertex_1_previous = circular_list.previous_item(vertex_1_rotation, old_edge)
+    vertex_1_previous_node = mesh.edge_nodes[vertex_1_previous]
+    vertex_1_next_key = circular_list.next_item(vertex_1_rotation, old_edge)
+
+    # 1.2 - Find the edges before and after the edge in the rotation of vertex_2.
+    vertex_2_rotation = list(vertex_trace(mesh, old_edge_node.vertex_2_key))
+    vertex_2_previous = circular_list.previous_item(vertex_2_rotation, old_edge)
+    vertex_2_previous_node = mesh.edge_nodes[vertex_2_previous]
+    vertex_2_next_key = circular_list.next_item(vertex_2_rotation, old_edge)
+
+    face_1_key = old_edge_node.face_1_key
+    face_2_key = old_edge_node.face_2_key
+
+    # 2 - Non-cofacial deletion.
+    if face_1_key != face_2_key:
+        # 2.1 - Update the face information.
+        # Create a new face.
+        new_face_key = mesh.create_face()
+
+        # Traverse face_1 and face_2, replace face_1 and face_2 with new_face.
+        for edge_key in list(
+            itertools.chain(face_trace(mesh, face_1_key), face_trace(mesh, face_2_key))
+        ):
+            edge_node = mesh.edge_nodes[edge_key]
+            if edge_node.face_1_key in (face_1_key, face_2_key):
+                edge_node.face_1_key = new_face_key
+            if edge_node.face_2_key in (face_1_key, face_2_key):
+                edge_node.face_2_key = new_face_key
+
+        # Delete face_1 and face_2.
+        mesh.delete_face(face_1_key)
+        mesh.delete_face(face_2_key)
+
+        # 2.2 - Update the edge information to delete the edge.
+        if vertex_1_previous_node.vertex_1_key == old_edge_node.vertex_1_key:
+            vertex_1_previous_node.vertex_1_next_key = vertex_1_next_key
+        else:
+            vertex_1_previous_node.vertex_2_next_key = vertex_1_next_key
+        if vertex_2_previous_node.vertex_1_key == old_edge_node.vertex_2_key:
+            vertex_2_previous_node.vertex_1_next_key = vertex_2_next_key
+        else:
+            vertex_2_previous_node.vertex_2_next_key = vertex_2_next_key
+
+        # Delete the edge.
+        mesh.delete_edge(old_edge)
+
+    # 3 - Cofacial deletion.
+    else:
+        # 3.1 - Update the edge information to delete the edge.
+        if vertex_1_previous_node.vertex_1_key == old_edge_node.vertex_1_key:
```

Comment:
suggestion: similar to the earlier suggestion, 2.2 is identical to 3.1

Reply:
See previous comment